### PR TITLE
Add Canon CLFs

### PR DESCRIPTION
**Pull request by doug-walker**
Adds the CLF for Canon Log 3 Cinema Gamut (Daylight) and its associated linear color space.  (The named transform for the log curve will need to wait for an update in the library to avoid needing to add an external LUT file.)

Thank you to the staff at Canon for their help in preparing these CLFs!

Note:  The CLF uses a Lut1D rather than a CameraLogTransform.  The Lut1D used has a normal domain, rather than a half-domain.  Hence the CLF supports normalized Canon Log 3 values only on a [0,1] domain.  This is very similar to what was done in the OCIO v1 ACES configs.

Signed-off-by: Doug Walker <doug.walker@autodesk.com>

### CODE REVIEW

**Comment by doug-walker** [comment]
I'm struggling to run the preflight.  I tried upgrading Poetry and reinstalling the dependencies but get the error:
```
An unexpected error has occurred: CalledProcessError: command: ('/Users/walkerdo/Library/Caches/pypoetry/virtualenvs/opencolorio-config-aces-3P9XfdaH-py3.9/bin/python', '-mvirtualenv', '/Users/walkerdo/.cache/pre-commit/repocveu0d_2/py_env-python3.8', '-p', 'python3.8')
return code: 1
expected return code: 0
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.8'
```
I'm running Python 3.9 and when I type "python" in the Poetry shell, it is 3.9.  So I don't understand why it is looking for Python 3.8.

**Comment by doug-walker** [comment]
Not sure why Poetry isn't able to do the preflight, so I just ran the the tools manually and made the necessary fixes.

**Comment by KelSolaar** [comment]
> I'm struggling to run the preflight. I tried upgrading Poetry and reinstalling the dependencies but get the error:
> 
> ```
> An unexpected error has occurred: CalledProcessError: command: ('/Users/walkerdo/Library/Caches/pypoetry/virtualenvs/opencolorio-config-aces-3P9XfdaH-py3.9/bin/python', '-mvirtualenv', '/Users/walkerdo/.cache/pre-commit/repocveu0d_2/py_env-python3.8', '-p', 'python3.8')
> return code: 1
> expected return code: 0
> stdout:
>     RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.8'
> ```
> 
> I'm running Python 3.9 and when I type "python" in the Poetry shell, it is 3.9. So I don't understand why it is looking for Python 3.8.

This is because the minimum Python language version supported by the pre-commit hooks is Python 3.8: https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/blob/main/.pre-commit-config.yaml#L6

I took the config from Colour but we could actually use Python 3.9 because OCIO follows the VFX Reference Platform and not the Python Scientific Ecosystem deprecation calendar.

File: opencolorio_config_aces/clf/transforms/canon/generate.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+"""
+*Canon* CLF Transforms Generation
+===============================
```
Comment (KelSolaar) [inline opencolorio_config_aces/clf/transforms/canon/generate.py:5]: **Minor**

*reStructuredText* tolerates longer underlines but not the shorter ones!

**Review by KelSolaar — COMMENTED** [review]
LGTM! (Except the small RST note)

**Comment by doug-walker** [comment]
I fixed the rst line.  Thanks for pointing me at the pre-commit yaml file.  I needed to change line 15 to 3.9 too but then it ran.

**Review by KelSolaar — APPROVED** [review]

**Pull request merged**